teamworkApp/lib/importData.py

```diff
@@ -1,26 +1,256 @@
+	"""Using data from the csv file given, insertstudent_data finds how many
+	students are in the file and places them into the student table"""
+	student_to_db = []
+
+	for student in student_dict.keys():
+		now = datetime.now().isoformat()
+		# TODO(Maeve): add functionality to get which team the student is on.
+			# This will require more functionality on the teams side
+		# currently name and username are the same
+		student_to_db += [(student, student_dict[student][0], now, student)]
+	return student_to_db
+
+def list_answer_data(student_dict):
+	"""Return the list of answer data corresponding to each student within
+	student_dict.
+	output: answer_to_db is a list of tuples holding information corresponding
+		to the columns within the answers table that is inserted in a future
+		function."""
+	# fetch the names of the students from the table
+	conn = sqlite3.connect(DB)
+	c = conn.cursor()
+	# get the primary keys for the students
+	c.execute('SELECT id FROM students;')
+	temp_keys = c.fetchall()
+	conn.commit()
+	conn.close()
+	primary_keys  = [temp_key[0] for temp_key in temp_keys]
+	answer_to_db = []
+	count = 0
+
+	# iterate through the students to find the enumerated responses to each
+		# question along with student_id and relevant timestamps
+	for student in student_dict.keys():
+		now                  = datetime.now().isoformat()
+		enummed_responses     = student_dict[student][2]
+		question_count       = 0
+
+		for response in enummed_responses:
+			answer_to_db += [
+			(response,
+			student_dict[student][0],
+			now,
+			primary_keys[count],
+			question_count)
+			]
+			question_count += 1
+		count += 1
+	return answer_to_db
+
+def list_style_data(student_dict):
+	"""Given a dictionary of student data, calculate and import style data
+	for each student.
+	Uses find_style to calculate styles."""
+	style_to_db = []
+	# fetch the names of the students from the table
 	conn = sqlite3.connect(DB)
 	c = conn.cursor()
+	# get the primary keys for the students
+	c.execute('SELECT id FROM students;')
+	temp_keys = c.fetchall()
+	conn.commit()
+	conn.close()
+
+	primary_keys = [temp_key[0] for temp_key in temp_keys]
 
-	c.execute('SELECT * FROM students')
+	count = 0
+	for student in student_dict.keys():
+		now               = datetime.now().isoformat()
+		expanded_responses = student_dict[student][1]
+		communicator, collaborator, challenger, contributor = find_style(expanded_responses)
+		# TODO: The commented out line below should be what we use in the
+			# future; however, evaluateAnswers has not been updated for the
+			# new schema (as of the time of writing this 10-28-2017) so we
+			# cannot use it yet
+		# (communicator, collaborator, challenger, contributor) = find_scores(primary_keys[count])
+		created_at   = student_dict[student][0]
+		student_id   = primary_keys[count]
+		style_to_db += [
+			(student_id,
+			communicator,
+			collaborator,
+			challenger,
+			contributor,
+			created_at,
+			now)
+			]
```

Comment:
For all of your list operations it would probably be cleaner to use list append

Reply:
Good call, updated